src/gui/dialogs/unit_recall.cpp

```diff
@@ -146,6 +146,14 @@ static std::string get_title_suffix(int side_num)
 	return msg.str();
 }
 
+bool unit_recall::unit_recall_default_compare(const unit_const_ptr first, const unit_const_ptr second)
+{
+	if (first->level() > second->level()) return true;
```

Comment:
Your custom sorting code is redundant. The listbox sorts the units by level *by default*, which overrides the order you set here. Please remove the custom sorting code, and only retain the change to the lambda in line 242.